### PR TITLE
RD-9485 - recursive parse triggeredBy

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/lumigo_tracer/event/event_trigger.py [hunk 1]
```diff
@@ -1,311 +1,35 @@
...
-
-TRIGGER_CREATION_TIME_KEY = "approxEventCreationTime"
-MESSAGE_ID_KEY = "messageId"
-MESSAGE_IDS_KEY = "messageIds"
-TOTAL_SIZE_BYTES = "totalSizeBytes"
-RECORDS_NUM = "recordsNum"
-MESSAGE_ID_TO_CHAINED_RESOURCE = "messageIdToChainResource"
-
-
-def parse_triggered_by(event: dict):  # type: ignore[no-untyped-def,type-arg]
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:21]: this file was split into many files - a file per trigger type.

File: src/test/unit/wrappers/http/test_sync_http_wrappers.py [hunk 2]
```diff
@@ -151,13 +151,18 @@ def lambda_test_function(event, context):
 
 def test_catch_file_like_object_sent_on_http(context, token):
     class A:
+        done = False
```
Comment (saartochner) [inline src/test/unit/wrappers/http/test_sync_http_wrappers.py:154]: This is a fix to a flaky test (work in some machines, and doesn't work in other)

File: src/test/unit/wrappers/http/test_sync_http_wrappers.py [hunk 3]
```diff
@@ -548,8 +553,8 @@ def mocked_begin(*args, **kwargs):
 
     @lumigo_tracer.lumigo_tracer(token=token)
     def lambda_test_function(event, context):
-        conn = http.client.HTTPConnection("www.google.com")
```
Comment (saartochner) [inline src/test/unit/wrappers/http/test_sync_http_wrappers.py:551]: fix another flaky - google sometimes throttle us

File: src/lumigo_tracer/event/event_trigger.py [hunk 4]
```diff
@@ -1,311 +1,38 @@
...
+from lumigo_tracer.event.trigger_parsing import EVENT_TRIGGER_PARSERS, INNER_MESSAGES_MAGIC_PATTERN
+from lumigo_tracer.event.trigger_parsing.event_trigger_base import TriggerType
+from lumigo_tracer.lumigo_utils import Configuration, get_logger
+
+
+def _recursive_parse_trigger_by(
+    message: Dict[Any, Any], parent_id: Optional[str], level: int
+) -> List[TriggerType]:
+    triggers = []
+    if level >= Configuration.chained_services_max_depth:
```
Comment (nirLumigo) [inline src/lumigo_tracer/event/event_trigger.py:13]: Great catch! 💪

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Nice! :)

File: src/lumigo_tracer/event/trigger_parsing/event_trigger_base.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            result["extra"] = extra
+        if from_message_ids:
+            result["fromMessageIds"] = from_message_ids
+        return result
+
+    @classmethod
+    def should_handle(cls, event: Dict[Any, Any]) -> bool:
+        try:
+            return cls._should_handle(event)
+        except Exception:
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/lumigo_tracer/event/trigger_parsing/event_trigger_base.py:52]: We talked about it: this is an accepted flow, we shouldn't log it

File: src/test/component/test_component.py [hunk 6]
```diff
@@ -172,11 +172,11 @@ def lambda_test_function(event, context):
     events = list(SpansContainer.get_span().spans.values())
     assert len(events) == 2
     # Single message.
-    assert events[0]["info"]["httpInfo"]["host"] == f"{region}.queue.amazonaws.com"
+    assert events[0]["info"]["httpInfo"]["host"] == f"sqs.{region}.amazonaws.com"
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/test/component/test_component.py:175]: yes. This is the old endpoint of SQS.
the docker's boto3 version was changed

**Review by nirLumigo — APPROVED** [review]

**Pull request merged**